proto/catalog.proto

```diff
@@ -45,6 +45,7 @@ message StreamSourceInfo {
   plan_common.EncodeType row_encode = 9;
   SchemaRegistryNameStrategy name_strategy = 10;
   optional string key_message_name = 11;
+  plan_common.StorageTableDesc upstream_table = 12;
```

Comment:
Btw you may want to have `Table` here instead (i.e. `message Table`.

In the future if you will support independent scaling of backfill cdc to the source, like in arrangement backfill, you will need replicated state table.

In that case `StorageTableDesc` is insufficient to construct the replicate state table.

Reply:
You may also derive table desc from the table definition iirc.

Reply:
I think @kwannoel is talking about you might need a state table to store the state like Kafka offset etc.? If so, that part is in the Executor proto. Here are protos of catalog

Reply:
Hi @kwannoel, do you mean the state table for persistent state of backfill? The storagetable here isn't for this purpose, it's information about the external table, e.g. schema, pk columns.
I will support persistent state in another PR.

Reply:
Nope, I meant the upstream state table.

Replicated state table is required for arrangement backfill, because we want to scale backfill independent to upstream mv.

With just a `TableDesc`, we can't construct this replicated state table.

> Hi @kwannoel, do you mean the state table for persistent state of backfill? The storagetable here isn't for this purpose, it's information about the external table, e.g. schema, pk columns.
I will support persistent state in another PR.

The replicated table will replicate the external table as mentioned here. Not the state of backfill.

Reply:
Did I understand it right? That `StorageTableDesc` corresponds to a `TableDesc` for `state_table`?

Reply:
If it does not apply in the case due to `mysql` cdc table being an external table, you may ignore it.

Reply:
> is it just a description for a table external to our system.

Yes.

> In arrangement backfill, the backfill executor and the mv executor may not be on the same CN. As a result, we need to replicate the records from upstream when they are not on the CN. Otherwise in-memory records will be lost.

Because in mv-on-mv scenario, snapshot read need to read from upstream mv, but in cdc backfill the snapshot is read from external database.